test/test_parser.cpp

```diff
@@ -3853,6 +3892,36 @@ TEST(Parser_OperatorIN)
     verify_query(test_context, t, "fav_item.name endswith[c] {'lk', 'EAL', 'GeS'}", 2);
     verify_query(test_context, t, "fav_item.name like {'*lk', '*zz*'}", 2);
 
+    // RHS is a param
+    using Vec = std::vector<Mixed>;
+    verify_query(test_context, t, "customer_id IN $0", {Vec{}}, 0);
```

Comment:
I manually verified that this *does* fail with the expected error if it is `{util::none}`, so it must be parsing this as an empty list rather than as null.